pkg/cache/sql/informer/factory/informer_factory.go

```diff
@@ -0,0 +1,134 @@
+/*
+Package factory provides an informer factory for the sql-based informer.
+*/
+package factory
+
+import (
+	"fmt"
+	"github.com/rancher/lasso/pkg/cache/sql/attachdriver"
+	db2 "github.com/rancher/lasso/pkg/cache/sql/db"
+	"github.com/rancher/lasso/pkg/cache/sql/encryption"
+	"github.com/rancher/lasso/pkg/cache/sql/informer"
+	sqlStore "github.com/rancher/lasso/pkg/cache/sql/store"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/tools/cache"
+	"os"
+	"sync"
+)
+
+// InformerFactory builds Informer instances and keeps a cache of instances it created
+type InformerFactory struct {
+	informerCreateLock sync.Mutex
+	wg                 wait.Group
+	dbClient           DBClient
+	stopCh             chan struct{}
+	encryptAll         bool
+
+	newInformer newInformer
+
+	cache map[schema.GroupVersionKind]*informer.Informer
+}
+
+type newInformer func(client dynamic.ResourceInterface, fields [][]string, gvk schema.GroupVersionKind, db sqlStore.DBClient, shouldEncrypt bool, namespace bool) (*informer.Informer, error)
+
+type DBClient interface {
+	informer.DBClient
+	sqlStore.DBClient
+	connector
+}
+
+type connector interface {
+	NewConnection() error
+}
+
+var defaultEncryptedResourceTypes = map[schema.GroupVersionKind]struct{}{
+	schema.GroupVersionKind{
+		Version: "v1",
+		Kind:    "Secret",
+	}: {},
+}
+
+const (
+	EncryptAllEnvVar = "CATTLE_ENCRYPT_CACHE_ALL"
+)
+
+func init() {
+	indexedFieldDBPath := db2.OnDiskInformerIndexedFieldDBPath
+	if os.Getenv(EncryptAllEnvVar) == "true" {
+		indexedFieldDBPath = ":memory:" // indexedFieldDBPath = "file::memory:?cache=shared"
+	}
+	attachdriver.Register("file:" + indexedFieldDBPath + "?cache=shared")
+}
+
+// NewInformerFactory returns an informer factory instance
+func NewInformerFactory() (*InformerFactory, error) {
+	m, err := encryption.NewManager()
+	if err != nil {
```

Comment:
@MbolotSuse there is at least one type that is always encrypted, Secrets and in the future tokens. I do think the types should be move up to steve and passed into lasso. At that time this can check if there are any types that will need encryption. For the present though there is no use case with some sort of encryption.

Reply:
Resolving since we will always use some form of encryption.